### ISSUE TITLE
`Y016` doesn't account for single member unions

### ISSUE THREAD

**Issue by tusharsadhwani** [PyCQA/flake8-pyi#484]
For code like:

```python
from typing import Union
x: Union[int, Union[int]]
```

Y016 should be raised but isn't.

**Comment by tusharsadhwani** [comment]
Same for `Y062` actually.

**Comment by tusharsadhwani** [comment]
> saying that the nested union is useless

More like a single member union `Union[X]` can always be replaced by `X`, I believe.

I'd be up for writing that rule. Y067? Does that work?

**Comment by JelleZijlstra** [comment]
I guess it's sort of redundant with Y037, which tells you to use `|` instead of `Union`. I wouldn't really be opposed to a PR here but I'm not convinced there's a serious problem our existing checks don't catch.
